bus-mapping/src/evm/opcodes/callvalue.rs

```diff
@@ -55,11 +64,28 @@ mod callvalue_tests {
         };
 
         // Get the execution steps from the external tracer
-        let block = crate::mock::BlockData::new_from_geth_data(
-            mock::new_single_tx_trace_code(&code).unwrap(),
-        );
+        let block: GethData = TestContext::<2, 1>::new(
```

Comment:
Most of the test setups in the for the `opcode::gen_associated_ops` in the bus-mapping are the same:
single transaction from A to B, where B has the specified code.  And the rest of the parameters are just default valid ones.  I'm thinking how could we simplify these tests to avoid the boilerplace.  I can think of two ideas:
1. In the opcodes module, add two testing functions:
```rust
fn account_0_code_account_1_no_code = |accs| {
                accs[0]
                    .address(address!("0x0000000000000000000000000000000000000010"))
                    .balance(Word::from(1u64 << 20))
                    .code(code);
                accs[1]
                    .address(address!("0x0000000000000000000000000000000000000000"))
                    .balance(Word::from(1u64 << 20));

                accs
            };
fn tx_from_0_to_1 = |mut txs, accs| {
                txs[0].to(accs[0].address).from(accs[1].address);
            };
```

And then in the tests do this:
```rust
        let block: GethData = TestContext::<2, 1>::new(
            None,
            account_0_code_account_1_no_code,
            tx_from_0_to_1,
            |block, _tx| block.number(0xcafeu64),
        )
        .unwrap()
        .into();
```

NOTE: the function names are just examples, there may be better names.

2. In the opcodes module, add one testing function:
```rust
const addr_a = address!("0x0000000000000000000000000000000000000000");
const addr_b = address!("0x0000000000000000000000000000000000000010");

fn test_ctx_a_calls_b() -> GethData {
TestContext::<2, 1>::new(
            None,
            |accs| {
                accs[0]
                    .address(addr_b)
                    .balance(Word::from(1u64 << 20))
                    .code(code);
                accs[1]
                    .address(addr_a)
                    .balance(Word::from(1u64 << 20));

                accs
            },
            |mut txs, accs| {
                txs[0].to(accs[0].address).from(accs[1].address);
            },
            |block, _tx| block.number(0xcafeu64),
        )
        .unwrap()
        .into();
}
```

And then in the test do this:
```rust
let block = test_ctx_a_calls_b();
```

And only when the default values for accounts / transaction / block need to be changed for the test, we use the `TestContext` builder directly.

What do you think?